crypto/keys/keybase_keyring.go

```diff
@@ -0,0 +1,511 @@
+func (kb keyringKeybase) Sign(name, passphrase string, msg []byte) (sig []byte, pub tmcrypto.PubKey, err error) {
+	info, err := kb.Get(name)
+	if err != nil {
+		return
+	}
+
+	var priv tmcrypto.PrivKey
+
+	switch info.(type) {
+	case localInfo:
+		linfo := info.(localInfo)
+		if linfo.PrivKeyArmor == "" {
+			err = fmt.Errorf("private key not available")
+			return
+		}
+
+		priv, err = cryptoAmino.PrivKeyFromBytes([]byte(linfo.PrivKeyArmor))
+		if err != nil {
+			return nil, nil, err
+		}
+
+	case ledgerInfo:
+		linfo := info.(ledgerInfo)
+		priv, err = crypto.NewPrivKeyLedgerSecp256k1Unsafe(linfo.Path)
+		if err != nil {
+			return
+		}
+
+	case offlineInfo, multiInfo:
+		_, err := fmt.Fprintf(os.Stderr, "Message to sign:\n\n%s\n", msg)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		buf := bufio.NewReader(os.Stdin)
+		_, err = fmt.Fprintf(os.Stderr, "\nEnter Amino-encoded signature:\n")
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// Will block until user inputs the signature
+		signed, err := buf.ReadString('\n')
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if err := cdc.UnmarshalBinaryLengthPrefixed([]byte(signed), sig); err != nil {
+			return nil, nil, errors.Wrap(err, "failed to decode signature")
+		}
+
+		return sig, info.GetPubKey(), nil
+	}
+
+	sig, err = priv.Sign(msg)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pub = priv.PubKey()
+	return sig, pub, nil
+}
+
+func (kb keyringKeybase) ExportPrivateKeyObject(name string, passphrase string) (tmcrypto.PrivKey, error) {
+	info, err := kb.Get(name)
+	if err != nil {
+		return nil, err
+	}
+
+	var priv tmcrypto.PrivKey
+
+	switch info.(type) {
+	case localInfo:
+		linfo := info.(localInfo)
+		if linfo.PrivKeyArmor == "" {
+			err = fmt.Errorf("private key not available")
+			return nil, err
+		}
+
+		priv, err = cryptoAmino.PrivKeyFromBytes([]byte(linfo.PrivKeyArmor))
+		if err != nil {
+			return nil, err
+		}
+
+	case ledgerInfo, offlineInfo, multiInfo:
+		return nil, errors.New("only works on local private keys")
+	}
+
+	return priv, nil
+}
+
+func (kb keyringKeybase) Export(name string) (armor string, err error) {
+	bz, err := kb.db.Get(string(infoKey(name)))
+	if err != nil {
+		return "", err
+	}
+	if bz.Data == nil {
+		return "", fmt.Errorf("no key to export with name %s", name)
+	}
+
+	return mintkey.ArmorInfoBytes(bz.Data), nil
+}
+
+// ExportPubKey returns public keys in ASCII armored format.
+// Retrieve a Info object by its name and return the public key in
+// a portable format.
+func (kb keyringKeybase) ExportPubKey(name string) (armor string, err error) {
+	bz, err := kb.Get(name)
+	if err != nil {
+		return "", err
+	}
+
+	if bz == nil {
+		return "", fmt.Errorf("no key to export with name %s", name)
+	}
+	return mintkey.ArmorPubKeyBytes(bz.GetPubKey().Bytes()), nil
+}
+
+func (kb keyringKeybase) Import(name string, armor string) (err error) {
+	bz, err := kb.Get(name)
+
+	if err == nil {
```

Comment:
delete this `if` statement

Reply:
the 'key not found' condition is the exception and triggers an early return